doc/tutorials/basic/build_image_classifier.md

```diff
@@ -0,0 +1,179 @@
+
+## Custom Components
+
+def get_label_view(fn, image: Image):
+    return get_plot(fn(image))
+
+def get_image_button(url, image_pane):
+    button = pn.widgets.Button(
+        width=100,
+        height=100,
+        stylesheets=[
+            f"button {{background-image: url({url});background-size: cover;}}"
+        ],
+    )
+    pn.bind(handle_example_click, button, url, image_pane, watch=True)
+    return button
+
+## Event Handlers
+
+def handle_example_click(event, url, image_pane):
+    image_pane.object = get_pil_image(url)
+
+
+def handle_file_upload(value, image_pane):
+    file = BytesIO(value)
+    image_pane.object = Image.open(file)
+
+def image_classification_interface(fn, examples):
+
+    ## State
+
+    # The image_view holds the *application state*
+    # The whole application is about updating or viewing it
+    # Should it be defined at the top?
+    image_view = pn.pane.Image(
+        get_pil_image(examples[0]),
+        height=IMAGE_DIM,
+        width=IMAGE_DIM,
+        fixed_aspect=False,
+        margin=0,
+    )
+
+    label_view = pn.pane.JSON()
+
+    ## Inputs
+
+    file_input = pn.widgets.FileInput(
+        accept=".png,.jpeg",
+    )
+    pn.bind(handle_file_upload, file_input, image_view, watch=True)
+    file_input_component = pn.Column("### Upload Image", file_input)
+
+    examples_input_component = pn.Column(
+        "### Examples", pn.Row(*(get_image_button(url, image_view) for url in examples))
+    )
+
+    ## Views
+
+    label_view = pn.Row(
+        pn.panel(pn.bind(get_label_view, fn=fn, image=image_view.param.object), defer_load=True, loading_indicator=True, height=IMAGE_DIM, width=IMAGE_DIM)
+    )
+
+    ## Layouts
+
+    input_component = pn.Column(
+        "# Input",
+        image_view,
+        file_input_component,
+        examples_input_component,
+        width=IMAGE_DIM,
+        margin=10,
+    )
+
+    output_component = pn.Column(
+        "# Output",
+        label_view,
+        width=IMAGE_DIM,
+        margin=10,
+    )
+    return pn.FlexBox(input_component, output_component)
+
+def predict(image: Image):
+    # Replace with your own image classification model
+    sleep(1.5)
+    a = random.uniform(0, 100)
+    b = random.uniform(0, 100 - a)
+    c = 100 - a - b
+    return {
+        "Wind Turbine": a,
+        "Solar Panel": b,
+        "Battery Storage": c,
+    }
+
+
+EXAMPLES = [
+    # Replace with your own examples
+    "https://upload.wikimedia.org/wikipedia/commons/thumb/b/ba/Windmills_D1-D4_%28Thornton_Bank%29.jpg/220px-Windmills_D1-D4_%28Thornton_Bank%29.jpg",
+    "https://upload.wikimedia.org/wikipedia/commons/thumb/9/90/Solar_cell.png/220px-Solar_cell.png",
+    "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a3/Overhead_View_of_Tehachapi_Energy_Storage_Project%2C_Tehachapi%2C_CA.png/220px-Overhead_View_of_Tehachapi_Energy_Storage_Project%2C_Tehachapi%2C_CA.png",
+]
+
+image_classification_interface(fn=predict, examples=EXAMPLES).servable()
+```
```

Comment:
I sort of agree but 😄 

The reasons why it is there is because

- I've agreed with Philipp that we try to put an e2e example into each section showcasing what has been learned. This app tries to do that.
- The ML App use case is hugely popular. See Gradio downloads. So I think we should have one.
- Having this relatively complicated example could maybe help us to make it easier in Panel such that it would not appear as a complicated recipe. We lack the high-level `Interface` for ML apps similarly to the `ChatInterface` we have built.
- I think its important to "give" users real examples in a tutorial that they can copy paste and start from.

What would you recommend doing @ahuang11 ?

Any comments @philippjfr

Reply:
Sometimes I see statements that its better to give the solution up front and break it down afterwards. That Is what I tried to do here. I.e. first the full app and then the *Break it down* section which has not yet been written.

I can do what you suggest @philippjfr . Its ok with me. But I would like more people to guide me first because this is a lot of work to change across the *Build a ...* tutorials.

Reply:
I wouldn't change this now to be clear. Let's focus on getting this in a mergeable state, which I think mostly involves me filling in a few more sections and then revisit once we've gathered some feedback.

Reply:
I think we are 50% to have an e2e tutorial. After that we will get a lot of feedback from HoloViz contributors and from users that will lead to refactoring.